shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Chain.hs

```diff
@@ -138,6 +140,23 @@ instance
   initialRules = []
   transitionRules = [chainTransition]
 
+chainChecks
+  :: Crypto crypto
+  => Natural
+  -> PParams
+  -> BHeader crypto
+  -> Maybe (PredicateFailure (CHAIN crypto))
```

Comment:
Just an idea: use `Except (PredicateFailure (CHAIN crypto)) ()` + `throwError`? (Or `Either`).

Reply:
yes, thanks, I should do something less unconventional.

how about `m ()` with the `MonadError (PredicateFailure (CHAIN crypto)) m` constraint? I just push this, you can see if it is less weird. :)

Reply:
Yeah, that's good :+1: